backend/groth16/bn254/mpcsetup/phase2.go

```diff
@@ -195,6 +245,17 @@ func (c *Phase2) Contribute() {
 	c.Parameters.G1.Delta.ScalarMultiplication(&c.Parameters.G1.Delta, &deltaBI)
 	c.Parameters.G2.Delta.ScalarMultiplication(&c.Parameters.G2.Delta, &deltaBI)
 
+	// Update BasisExpSigma with δ
+	// TODO: Is it sound to use the same δ for all basis?
+	for i := 0; i < len(c.Parameters.G1.BasisExpSigma); i++ {
+		for j := 0; j < len(c.Parameters.G1.BasisExpSigma[i]); j++ {
+			c.Parameters.G1.BasisExpSigma[i][j].ScalarMultiplication(&c.Parameters.G1.BasisExpSigma[i][j], &deltaBI)
+		}
+	}
+
+	// Update GRootSigmaNeg using δ⁻¹
+	c.Parameters.G2.GRootSigmaNeg.ScalarMultiplication(&c.Parameters.G2.GRootSigmaNeg, &deltaInvBI)
+
```

Comment:
aren't we supposed to verify the ratio of this in `verifyPhase2` function?

Reply:
@hussein-aitlahcen yep that's correct! Sorry for the slow reply, I've been away for the week! I'll push an update for this in an hour or so!

Reply:
I've actually patched my fork with your change, rebased on top of main and added the verification if you want to absorb it https://github.com/hussein-aitlahcen/gnark/commit/9d859afe4c14da0e91232fa078fe4b65c79cf667. I also fixed the cloning of the phase2 which was missing the pedersen key/marshalling and was able to do a full mpc round using this.

Reply:
@hussein-aitlahcen Thanks for doing that! I've just committed your changes to this PR. Although it might be nice to eventually upstream the marshaling stuff.

I'm getting some issues with verification though it seems.

Reply:
What error are you getting? I'm able to do phase2 init + contrib + verification. Did it a couple of time.

Reply:
Ah I manually recreated your fix but copy and pasted wrong. Just cherry picked and now things seem to be working!